feat(core): add schematic to escape block syntax characters

crisbeto opened the pull request:
#51891 introduces a new syntax that assigns a new meaning to the `@` and `}` characters in Angular templates. This is problematic for existing apps which may have the characters in their templates already, because it can lead to syntax errors.

These changes add an `ng update` schematic that will replace any usages of the special characters with their HTML entities.

atscott commented on packages/core/schematics/migrations.json:
```diff
@@ -9,6 +9,11 @@
       "factory": "./migrations/guard-and-resolve-interfaces/bundle"
+    },
+    "migration-v17-block-template-entities": {
+      "version": "17.0.0-0",
```
Is this version intentional?

crisbeto replied:
Yeah, it allows the migration to run against prerelase versions.

atscott commented on packages/core/schematics/migrations/block-template-entities/util.ts:
```diff
@@ -0,0 +1,170 @@
+}
+
+/**
+ * Analyzies a source file to find file that need to be migrated and the text ranges within them.
```
```suggestion
 * Analyzes a source file to find file that need to be migrated and the text ranges within them.
```

Or is Analyzies the cutesy version of Analyses? :)

AndrewKushnir commented on packages/core/schematics/migrations/block-template-entities/util.ts:
```diff
@@ -0,0 +1,170 @@
+      continue;
+    }
+
+    // Note: we have a utility to resolve the Angular decorators from a class declaration already.
+    // We don't use it here, because it requires access to the type checker which makes it more
+    // time-consuming to run internally.
```
Can we move this code to one of the shared utils lib, so that we can use it in other future migrations?

crisbeto replied:
I think that in most migrations we'd want to use the existing utility, but for this one I wanted to go with this one because it allows it to run quicker in g3.

crisbeto committed:
feat(core): add schematic to escape block syntax characters

#51891 introduces a new syntax that assigns a new meaning to the `@` and `}` in Angular templates. This is problematic for existing apps which may have the characters in their templates already, because it can lead to syntax errors.

These changes add an `ng update` schematic that will replace any usages of the special characters with their HTML entities.

dylhunn commented:
This PR was merged into the repository by commit c7127b98b555449f99e24a81c828ab7b1c6c4a4e.
